lib/livebook_web/live/session_live/fly_runtime_component.ex

```diff
@@ -610,18 +748,40 @@ defmodule LivebookWeb.SessionLive.FlyRuntimeComponent do
         }
   end
 
-  defp specs_changeset(config, attrs \\ %{}) do
-    defaults = %{
+  defp config_secret_names(hub_secrets) do
+    names =
+      for %{name: name} <- hub_secrets,
+          String.starts_with?(name, @config_secret_prefix),
+          do: name
+
+    Enum.sort(names)
+  end
+
+  defp load_config_defaults(socket) do
+    config_defaults = socket.assigns.config_defaults
+
+    socket
+    |> assign(
+      token: config_defaults["token"],
+      app_name: config_defaults["app_name"],
+      specs_changeset: specs_changeset(config_defaults)
+    )
+    |> load_org_and_regions()
+    |> load_app()
+  end
+
+  defp specs_changeset(attrs \\ %{}) do
+    docker_tags = Enum.map(Livebook.Config.docker_images(), & &1.tag)
+
+    data = %{
       cpu_kind: "shared",
       cpus: 1,
       memory_gb: 1,
       gpu_kind: nil,
       gpus: nil,
-      docker_tag: Livebook.Config.docker_images() |> hd() |> Map.fetch!(:tag)
+      docker_tag: hd(docker_tags)
     }
```

Comment:
I wonder if we should create an actual changeset for it now? It will also help us unit test the serialization and deseriaization logic.

Reply:
A single changeset is tricky, in particular with volumes. We know the valid volume options only once the app loads. More importantly, in the volume section we have a "Create volume" form, and since we can't have nested forms, it means we can't really have a single form for all the config fields.